### PR TITLE
DBZ-6731 Document blocking snapshot signal configuration

### PR DESCRIPTION
**Pull request by mfvitale**
related to: https://github.com/debezium/debezium/pull/4741

### CODE REVIEW

File: documentation/modules/ROOT/pages/configuration/signalling.adoc [hunk 1]
```diff
@@ -231,9 +243,21 @@ Currently {prodname} supports the `incremental` and `blocking` types.
 | An array of comma-separated regular expressions that match the fully-qualified names of the tables to include in the snapshot. +
 Specify the names by using the same format as is required for the xref:{context}-property-signal-data-collection[signal.data.collection] configuration option.
 
-|`additional-condition`
+|`[.line-through]#additional-condition#`
+|_N/A_
+| An optional string that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
+ +
+  [NOTE]
```
Comment (mfvitale) [inline documentation/modules/ROOT/pages/configuration/signalling.adoc:250]: @roldanbob can you help me identify why this is not rendered as a `note` box? I saw it is used also in other pages.

File: documentation/modules/ROOT/pages/configuration/signalling.adoc [hunk 2]
```diff
@@ -231,9 +243,21 @@ Currently {prodname} supports the `incremental` and `blocking` types.
...
+  ====
+  This property is deprecated and should be replaced by additional-conditions.
+  ====
+
+|`additional-conditions`
 |_N/A_
-| An optional string that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot.
+| An optional array of additional condition that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
+Each additional condition is an object with `data-collection` and `filter` properties. You can now specify different filters for different data collection. +
+* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
```
Comment (mfvitale) [inline documentation/modules/ROOT/pages/configuration/signalling.adoc:259]: Also this bullet list is not working. Any idea?

File: documentation/modules/ROOT/pages/configuration/signalling.adoc (hunk 1 above)
Comment (mfvitale) [reply documentation/modules/ROOT/pages/configuration/signalling.adoc:250]: Got it! Many thanks!

File: documentation/modules/ROOT/pages/configuration/signalling.adoc (hunk 2 above)
Comment (roldanbob) [reply documentation/modules/ROOT/pages/configuration/signalling.adoc:259]: Same solution as above

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc [hunk 3]
```diff
@@ -24,9 +24,22 @@ See the next section for more details.
...
+| An optional string that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
+
+[NOTE]
+====
+This property is deprecated and should be replaced by additional-conditions.
+====
+
+|`additional-conditions`
+|_N/A_
+| An optional array of additional condition that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc:38]:
```suggestion
| An optional array of additional conditions that specifies criteria that the connector evaluates to designate a subset of records to include in a snapshot. +
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc [hunk 4]
```diff
@@ -24,9 +24,22 @@ See the next section for more details.
...
+Each additional condition is an object with `data-collection` and `filter` properties. You can now specify different filters for different data collection.
+
+* The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
+* The `filter` property will have the same value used in the old property for the `incremental snapshot` and the value used in the `snapshot.select.statement.overrides` for the `blocking snapshot`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc:42]:
```suggestion
Each additional condition is an object that specifies the criteria for filtering the data that an ad hoc snapshot captures.
You can set the following parameters for each additional condition:
`data-collection`:: The fully-qualified name of the {data-collection} that the filter applies to.
You can apply different filters to each {data-collection}. 
`filter`:: Specifies column values that must be present in a database record for the snapshot to include it, for example,  `"color='blue'"`. +
 +
The values that you assign to the `filter` parameter are the same types of values that you might specify in the `WHERE` clause of `SELECT` statements when you set the `snapshot.select.statement.overrides` property for a blocking snapshot.
In earlier {prodname} releases, an explicit `filter` parameter was not defined for snapshot signals; instead, filter criteria were implied by the values that were specified for the now deprecated `additional-condition` parameter. 
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc [hunk 5]
```diff
@@ -37,30 +50,30 @@ Key = `test_connector`
...
 
-{prodname} uses the `additional-condition` field to select a subset of a {data-collection}'s content.
+{prodname} uses the `additional-conditions` field to select a subset of a {data-collection}'s content.
 
 Typically, when {prodname} runs a snapshot, it runs a SQL query such as:
 
 `SELECT * FROM _<tableName>_ ....`
 
-When the snapshot request includes an `additional-condition`, the `additional-condition` is appended to the SQL query, for example:
+When the snapshot request includes an `additional-conditions`, the `data-collection` and `filter` property of the `additional-conditions` is appended to the SQL query, for example:
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc:61]:
```suggestion
When the snapshot request includes an `additional-conditions` property, the `data-collection` and `filter` parameters of the  property are appended to the SQL query, for example:
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-sql.adoc [hunk 6]
```diff
@@ -84,16 +84,17 @@ Currently, the only valid option is the default value, `incremental`. +
 If you do not specify a value, the connector runs an incremental snapshot.
 
 |6
-|`additional-condition`
-|An optional string, which specifies a condition based on the column(s) of the {data-collection}(s), to capture a
-subset of the contents of the {data-collection}s.
-For more information about the `additional-condition` parameter, see xref:{context}-incremental-snapshots-additional-condition[].
+|`additional-conditions`
+| An optional array of additional condition that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-sql.adoc:88]:
```suggestion
| An optional array that specifies a set of additional conditions that the connector evaluates to determine the subset of records to include in a snapshot. +
```

File: documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-sql.adoc [hunk 7]
```diff
@@ -84,16 +84,17 @@ Currently, the only valid option is the default value, `incremental`. +
...
+Each additional condition is an object with `data-collection` and `filter` properties. You can now specify different filters for different data collection. +
```
Comment (roldanbob) [inline documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-sql.adoc:89]:
```suggestion
Each additional condition is an object with `data-collection` and `filter` properties. 
You can specify different filters for each data collection. +
```

File: documentation/modules/ROOT/pages/configuration/signalling.adoc [hunk 8]
```diff
@@ -231,9 +244,22 @@ Currently {prodname} supports the `incremental` and `blocking` types.
...
+This property is deprecated and should be replaced by additional-conditions.
+====
+
+|`additional-conditions`
+|_N/A_
+| An optional array of additional condition that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
+Each additional condition is an object with `data-collection` and `filter` properties. You can now specify different filters for different data collection.
+
+ * The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
+ * The `filter` property will have the same value used in the old property for the `incremental snapshot` and the value used in the `snapshot.select.statement.overrides` for the `blocking snapshot`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/configuration/signalling.adoc:262]:
```suggestion
Each additional condition is an object that specifies the criteria for filtering the data that an ad hoc snapshot captures.
You can set the following parameters for each additional condition:

`data-collection`:: The fully-qualified name of the {data-collection} that the filter applies to.
You can apply different filters to each {data-collection}. 

`filter`:: Specifies column values that must be present in a database record for the snapshot to include it, for example,  `"color='blue'"`. +
 +
The values that you assign to the `filter` parameter are the same types of values that you might specify in the `WHERE` clause of `SELECT` statements when you set the `snapshot.select.statement.overrides` property for a blocking snapshot.
In earlier {prodname} releases, an explicit `filter` parameter was not defined for snapshot signals; instead, filter criteria were implied by the values that were specified for the now deprecated `additional-condition` parameter. 
```

File: documentation/modules/ROOT/pages/configuration/signalling.adoc [hunk 9]
```diff
@@ -178,12 +178,25 @@ Currently {prodname} supports the `incremental` and `blocking` types.
...
+====
+
+|`additional-conditions`
 |_N/A_
-| An optional string that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot.
+| An optional array of additional condition that specifies a condition that the connector evaluates to designate a subset of records to include in a snapshot. +
+  Each additional condition is an object with `data-collection` and `filter` properties. You can now specify different filters for different data collection.
+
+ * The `data-collection` property is the fully-qualified name of the data collection for which the filter will be applied.
+ * The `filter` property will have the same value used in the old property for the `incremental snapshot` and the value used in the `snapshot.select.statement.overrides` for the `blocking snapshot`.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/configuration/signalling.adoc:199]:
```suggestion
Each additional condition is an object that specifies the criteria for filtering the data that an ad hoc snapshot captures.
You can set the following parameters for each additional condition:

You can set the following parameters for each additional condition:

`data-collection`:: The fully-qualified name of the {data-collection} that the filter applies to.
You can apply different filters to each {data-collection}. 
`filter`:: Specifies column values that must be present in a database record for the snapshot to include it, for example,  `"color='blue'"`. +
 +
The values that you assign to the `filter` parameter are the same types of values that you might specify in the `WHERE` clause of `SELECT` statements when you set the `snapshot.select.statement.overrides` property for a blocking snapshot.
In earlier {prodname} releases, an explicit `filter` parameter was not defined for snapshot signals; instead, filter criteria were implied by the values that were specified for the now deprecated `additional-condition` parameter. 
```

**Review by roldanbob — CHANGES_REQUESTED** [review]
@mfvitale I got a little confused with this one, because I wasn't sure how to modify the language that mentioned that the `filter` property had "the same value used in other contexts, which isn't strictly correct. I reworked my original edits and I think that I've figured out how to express what you meant. Let me know if you disagree or have any questions.
The one change that might still be useful is to enforce consistency when we mention `additional-conditions` properties vs. parameters.  I started to use _properties_ to talk about `additional-conditions`, but then use _parameters_ when talking about the fields within the `additional-conditions` object, but I stopped after finding so many instances of different usage, and I began to think that maybe I'm trying to apply create a distinctions that isn't necessary?  WDYT?

File: documentation/modules/ROOT/pages/configuration/signalling.adoc (hunk 9 above)
Comment (mfvitale) [reply documentation/modules/ROOT/pages/configuration/signalling.adoc:199]: > You can set the following parameters for each additional condition:

Also these are still properties

> The values that you assign to the `filter` parameter are the same types of values that you might specify in the `WHERE` clause of `SELECT` statements when you set the `snapshot.select.statement.overrides` property for a blocking snapshot.
	In earlier {prodname} releases, an explicit `filter` parameter was not defined for snapshot signals; instead, filter criteria were implied by the values that were specified for the now deprecated `additional-condition` parameter.

This is not really true. If you are doing an incremental snapshot you can now use the filter to put what should be used in the where clause (the same used in the `additional-condition`) with the plus that now you can specify multiple filters. 

For the blocking snapshot the filters has a different meaning, you need to pass the full query statement as you can do for the initial snapshot using the `snapshot.select.statement.overrides`

Is it more clear now?

**Comment by jpechane** [comment]
@mfvitale @roldanbob Applied, thanks

**Comment by roldanbob** [comment]
> @mfvitale @roldanbob Applied, thanks

@jpechane Per his [last comment](https://github.com/debezium/debezium/pull/4769#discussion_r1290997185), I think that @mfvitale had some concerns and was looking for a further edit. I was working on something yesterday, but I had not yet added it here. 
Should I go ahead an submit a new PR for 6731 with changes to address those concerns.

**Comment by jpechane** [comment]
@roldanbob I added a commit that tried to address them - https://github.com/debezium/debezium/pull/4769/commits/63fed8b8ffaa703c34b5213e87fa78a500706624

Feel free to change the text s necessary

**Comment by roldanbob** [comment]
> @roldanbob I added a commit that tried to address them - [63fed8b](https://github.com/debezium/debezium/commit/63fed8b8ffaa703c34b5213e87fa78a500706624)

Ah, ok. I missed that. I'll take a look. Thanks, @jpechane!

**Pull request merged**